src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceKnownTypeTests.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.ServiceModel;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+
+public static class ServiceKnownTypeTests
+{
+    [Fact]
+    [OuterLoop]
+    public static void ServiceKnownType_DataContract_AttrOnMethod_Test()
+    {
+        BasicHttpBinding binding = new BasicHttpBinding();
+        EndpointAddress endpointAddress = new EndpointAddress(Endpoints.HttpBaseAddress_Basic);
+        ChannelFactory<IServiceKnownTypeTest_AttrOnMethod> factory = new ChannelFactory<IServiceKnownTypeTest_AttrOnMethod>(binding, endpointAddress);
+        IServiceKnownTypeTest_AttrOnMethod serviceProxy = factory.CreateChannel();
+
+        try
+        {
+            var widget = new Widget0();
+            widget.Id = "1";
+            widget.Catalog = "widget";
+
+            var input = new object[] { widget };
+            var response = serviceProxy.EchoItems(input);
+
+            Assert.Equal(widget.Id, ((Widget)response[0]).Id);
+        }
```

Comment:
As the last thing before the Finally add some lines to attempt closing of any communication objects...

// *** CLEANUP *** \\
            ((ICommunicationObject)serviceProxy).Close();
            factory.Close();